### PR TITLE
Init schema and skeleton API setup

### PR DESCRIPTION
**Pull request by sky159321**
## Description
Initial schema setup + skeleton API + removed some unused code as of now.

## Questions for Reviewers
Send it

## Testing
N/A, will add unit tests soon.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -54,13 +48,7 @@ app.use(passport.initialize());
...
-
-intializePassport(
-  passport,
-  getProfileByUsername,
-  getProfileById,
-  getProfileByEmail
-);
+intializePassport(passport, getProfileById, getProfileByEmail);
 
 const server = http.createServer(app);
```
Comment (garvsl) [inline index.ts:53]: comment out

File: index.ts [hunk 2]
```diff
@@ -78,6 +66,7 @@ io.use((socket: any, next: any) => {
 
 export let onlineUsers: any = {};
```
Comment (garvsl) [inline index.ts:69]: comment out

File: index.ts [hunk 3]
```diff
@@ -78,6 +66,7 @@ io.use((socket: any, next: any) => {
...
 
+// @TODO: cache "onlineUsers"
 io.on("connection", async (socket: any) => {
```
Comment (garvsl) [inline index.ts:72]: comment out anything io related for now

File: passport.ts [hunk 4]
```diff
@@ -14,16 +13,16 @@ export default function intializePassport(
...
 
-    if (!isInputEmail) {
-      user = await getProfileByUsername(username);
-    } else {
-      user = await getProfileByEmail(username);
+    if (!regex.test(username)) {
+      return done(null, false, {
+        message: "Not an email, please try again",
+      });
     }
```
Comment (garvsl) [inline passport.ts:21]: probably should  change the name of paramtered from username to email

File: prisma/schema.prisma [hunk 5]
```diff
@@ -11,92 +11,83 @@ datasource db {
...
+  gender         Gender?
+  weightLb       Int?
+  heightFoot     Int?
+  heightInch     Int?
+  squatLb        Int?
+  deadLiftLb     Int?
+  benchPressLb   Int?
+  workoutPattern Json?
+  archetype      Archetype?
+  yoe            Int?
```
Comment (garvsl) [inline prisma/schema.prisma:30]: nice

File: queries/partnerQueries.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      },
+    });
+
+    const matchingFriends = user?.partners.filter((partner) =>
+      partner.name.toLowerCase().includes(partialUsername.toLowerCase())
+    );
+    return matchingFriends;
+  } catch (e) {
+    console.log(e);
+  }
```
Comment (garvsl) [inline queries/partnerQueries.ts:95]: niceey

File: prisma/schema.prisma [hunk 7]
```diff
@@ -11,92 +11,83 @@ datasource db {
 }
 
 model User {
-  id             Int      @id @default(autoincrement())
-  username       String   @unique
-  email          String   @unique
-  password       String
-  picture        String
```
Comment (garvsl) [inline prisma/schema.prisma:18]: probably still need profile picture

File: routes/login.ts [hunk 8]
```diff
@@ -21,38 +16,38 @@ sgMail.setApiKey(process.env.SENDGRID_API as string);
 const client = new OAuth2Client(process.env.CLIENTID);
 const router = Router();
 
-router.get("/", async (req:any, res:any) => {
+router.get("/", async (req: any, res: any) => {
   if (req.session.user) {
-    const user = await getProfileById(Number(req.session.user));
+    const user = await getProfileById(req.session.user);
```
Comment (garvsl) [inline routes/login.ts]: wouldnt this need to be the uuid? Im pretty sure the req.session.user is the users id

File: routes/login.ts [hunk 9]
```diff
@@ -78,13 +72,13 @@ router.post("/googleOauth", async (req:any, res:any, next) => {
 
         const user: any = await getProfileByEmail(email);
 
-        req.logIn(user, async (err:any) => {
+        req.logIn(user, async (err: any) => {
           if (err) {
             return next(err);
           }
 
           req.session.user = user.id.toString();
```
Comment (garvsl) [inline routes/login.ts]: we could change this to be the uuid

File: routes/login.ts [hunk 10]
```diff
@@ -21,38 +16,38 @@ sgMail.setApiKey(process.env.SENDGRID_API as string);
...
     }
     if (!user) {
       return res.status(401).json({ message: "This user does not exist!" });
     }
-    req.logIn(user, async (err:any) => {
+    req.logIn(user, async (err: any) => {
       if (err) {
         return next(err);
       }
       req.session.user = user.id.toString();
```
Comment (garvsl) [inline routes/login.ts]: we could change this to be the uuid

File: routes/login.ts [hunk 11]
```diff
@@ -99,24 +93,16 @@ router.post("/googleOauth", async (req:any, res:any, next) => {
...
+  const user = await getProfileByEmail(email);
 
   if (user !== null && user && user.registeredWith !== "google") {
     const resetToken = Math.floor(100000000 + Math.random() * 900000000);
-    await updateUserResetToken(Number(user.id), resetToken.toString());
+    await updateUserResetToken(user.id, resetToken.toString());
     const msg = {
       to: `${user.email}`,
-      from: "crowdwrap@gmail.com",
+      from: "lwb@gmail.com",
```
Comment (garvsl) [inline routes/login.ts:105]: would need to update the api for this, and also create an email with that name

File: routes/partners.ts [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+import { onlineUsers } from "../index";
+import {
+  addPartner,
+  removePartner,
+  getPartnersByPartialUsername,
+} from "../queries/partnerQueries";
+import { getProfileById } from "../queries/profileQueries";
+const router = Router();
+
+// A lot of scalablity problems with the friendlist, for example the notificatins. There would be a problem if there were too many active users (Redis could solve)
```
Comment (garvsl) [inline routes/partners.ts]: why dis still here :sob:

**Comment by garvsl** [comment]
So when they register we will set their userId to a random uuid?

File: routes/login.ts (hunk 11 above)
Comment (sky159321) [reply routes/login.ts:105]: which api ?

**Comment by sky159321** [comment]
> So when they register we will set their userId to a random uuid?

we could either generate, then save. Or let the DB generate, then set the session's userId to be the generated uuid from DB

**Comment by garvsl** [comment]
> > So when they register we will set their userId to a random uuid?
> 
> we could either generate, then save. Or let the DB generate, then set the session's userId to be the generated uuid from DB

Second option sounds better, what do you prefer?

**Review by garvsl — APPROVED** [review]

**Pull request merged**